test_eos-phone-home.py

```diff
@@ -0,0 +1,159 @@
+                            ensure=True)
+
+    ssn = tmpdir.join('sys', 'class', 'endless_mfgdata', 'entries', 'SSN')
+    ssn.write('  serial  ', ensure=True)
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo('vendor1', 'model1', 'serial')
+
+
+def test_get_product_info_arm_missing_serial(tmpdir, phone_home):
+    compatible = tmpdir.join('proc', 'device-tree', 'compatible')
+    compatible.write_binary(b'vendor1,model1\0vendor2,model2\0',
+                            ensure=True)
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo('vendor1', 'model1', None)
+
+
+def test_get_product_info_arm_missing_dt(tmpdir, phone_home):
+    ssn = tmpdir.join('sys', 'class', 'endless_mfgdata', 'entries', 'SSN')
+    ssn.write('  serial  ', ensure=True)
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo(None, None, 'serial')
+
+
+def test_get_product_info_x86(tmpdir, phone_home):
+    scdi = tmpdir.join('sys', 'class', 'dmi', 'id')
+    scdi.join('sys_vendor').write('  vendor', ensure=True)
+    scdi.join('product_name').write(' product')
+    scdi.join('product_serial').write('serial  ')
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo('vendor', 'product', 'serial')
+
+
+def test_get_product_info_x86_missing_vendor(tmpdir, phone_home):
+    scdi = tmpdir.join('sys', 'class', 'dmi', 'id')
+    scdi.join('product_name').write(' product', ensure=True)
+    scdi.join('product_serial').write('serial  ')
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo(None, 'product', 'serial')
+
+
+def test_singleboot(tmpdir, phone_home):
+    cmdline = ''
+    tmpdir.join('proc', 'cmdline').write(cmdline, ensure=True)
+
+    assert phone_home._get_dualboot() is False
+    assert phone_home._get_live() is False
+
+
+def test_dualboot(tmpdir, phone_home):
+    cmdline = '''
+    endless.image.device=UUID=blahblah
+    endless.image.path=/endless/endless.img
+    '''.strip()
+    tmpdir.join('proc', 'cmdline').write(cmdline, ensure=True)
+
+    assert phone_home._get_dualboot() is True
+    assert phone_home._get_live() is False
+
+
+def test_live(tmpdir, phone_home):
+    cmdline = '''
+    endless.image.device=UUID=blahblah
+    endless.image.path=/endless/endless.img
+    endless.live_boot
+    '''.strip()
+    tmpdir.join('proc', 'cmdline').write(cmdline, ensure=True)
+
+    assert phone_home._get_dualboot() is False
+    assert phone_home._get_live() is True
+
+
+def test_count_initial(tmpdir, phone_home):
+    assert phone_home._get_count() == 0
+
+
+def test_count_subsequent(tmpdir, phone_home):
+    tmpdir.join('var', 'lib', 'eos-phone-home', 'count').write('5',
+                                                               ensure=True)
+    assert phone_home._get_count() == 5
+
+
+def test_build_activation_request_empty(tmpdir, phone_home):
```

Comment:
Nit: Test name is unclear about what the expected outcome of the test should be after building an empty request

Reply:
I have renamed this test to `test_build_activation_request_when_fields_cannot_be_determined`